src/beautifiers/clang-format.coffee

```diff
@@ -0,0 +1,60 @@
+###
+Requires clang-format (https://clang.llvm.org)
+###
+
+"use strict"
+Beautifier = require('./beautifier')
+path = require('path')
+fs = require('fs')
+
+module.exports = class ClangFormat extends Beautifier
+
+  name: "clang-format"
+
+  options: {
+    "C++": true
+    "C": true
+    "Objective-C": true
+  }
+
+  ###
+    Dump contents to a given file
+  ###
+  dumpToFile: (name = "atom-beautify-dump", contents = "") ->
+    return new Promise((resolve, reject) =>
+      fs.open(name, "w", (err, fd) =>
+        @debug('dumpToFile', name, err, fd)
+        return reject(err) if err
+        fs.write(fd, contents, (err) ->
+          return reject(err) if err
+          fs.close(fd, (err) ->
+            return reject(err) if err
+            resolve(name)
+          )
+        )
+      )
+    )
+
+  beautify: (text, language, options) ->
+    return new @Promise((resolve, reject) ->
+      editor = atom.workspace.getActiveTextEditor()
```

Comment:
I think the current solution is acceptable. I will have to consider giving the Beautifier class access to the file path and not only its contents.

You could change:

```coffee
editor = atom.workspace.getActiveTextEditor()
```

to 

```coffee
editor = atom?.workspace?.getActiveTextEditor()
```

Such that it does not error if it does not run within Atom, and `editor` will then be `null`.

This is because beautification testing may run from outside of Atom and I want to make sure it works (does not crash) and throws errors if it does require Atom (which this does).

Thanks!